src/main/kotlin/no/nav/syfo/kartleggingssporsmal/application/KartleggingssporsmalService.kt

```diff
@@ -136,14 +136,40 @@ class KartleggingssporsmalService(
         } else {
             val statusendring = KartleggingssporsmalKandidatStatusendring(status = KandidatStatus.SVAR_MOTTATT, svarAt = svarAt)
             val kandidat = existingKandidat.registrerStatusEndring(statusendring)
-            val createdStatusendring = kartleggingssporsmalRepository.createKandidatSvar(kandidat, statusendring)
+            val createdStatusendring = kartleggingssporsmalRepository.createKandidatStatusendring(kandidat, statusendring)
             kartleggingssporsmalKandidatProducer.send(kandidat, createdStatusendring)
                 .map { kandidat ->
                     kartleggingssporsmalRepository.updatePublishedAtForKandidatStatusendring(createdStatusendring)
                 }
         }
     }
 
+    suspend fun registrerFerdigBehandlet(
+        uuid: UUID,
+        veilederident: String,
+    ): KartleggingssporsmalKandidat {
+        val existingKandidat = kartleggingssporsmalRepository.getKandidat(uuid)
+
+        if (existingKandidat?.status != KandidatStatus.SVAR_MOTTATT) {
+            throw IllegalArgumentException("Kandidat finnes ikke, eller er allerede ferdig behandlet")
+        } else {
```

Comment:
Hadde tenkt at rekkefølgen man kan gjøre disse vurderingene i er forretningslogikk og burde ligge i domenet.

Dette kunne for eksempel sett ut på denne måten 👇
Så hadde `registrerFerdigbehandlet()` gjort sjekken om at svar er nødt å være mottatt for at overgangen til `FERDIGBEHANDLET` skal være gyldig.
```suggestion
        val existingKandidat = kartleggingssporsmalRepository.getKandidat(uuid)

        val ferdigbehandletKandidat = existingKandidat.registrerStatusendring(status)
```

Dette er også noe vi eventuelt kan refaktorere senere 🤔